src/rgw/rgw_op.cc

```diff
@@ -3739,6 +3740,7 @@ void RGWPutObj::execute(optional_yield y)
   rgw_placement_rule *pdest_placement = &s->dest_placement;
 
   if (multipart) {
+    s->trace->SetTag(tracing::UPLOAD_ID, multipart_upload_id);
```

Comment:
do we make sure s->trace is always not null ?

Reply:
> do we make sure s->trace is always not null ?

no. 
we can assume the span isn't null. we create the span without any conditions.
in case tracing is disabled or jager isn't compiled. there will be a stub span.